Ghidra/Features/Base/src/main/java/ghidra/util/data/DataTypeParser.java

```diff
@@ -561,28 +573,53 @@ int getElementCount() {
 	}
 
 	private static class PointerSpecPiece implements DtPiece {
+		int shiftOffset = 0;
 		int pointerSize = -1;
-
+		
 		PointerSpecPiece(String piece) throws InvalidDataTypeException {
-			if (!piece.startsWith("*")) {
+			if (!piece.contains("*")) {
 				throw new InvalidDataTypeException("invalid pointer specification: " + piece);
 			}
-			if (piece.length() == 1) {
-				return;
-			}
-			try {
-				pointerSize = Integer.parseInt(piece.substring(1));
-			}
-			catch (NumberFormatException e) {
-				throw new InvalidDataTypeException("invalid pointer specification: " + piece);
+			
```

Comment:
It may be nice to put a comment with an example of what is being parsed here. Just a general example of what the shifted pointer syntax would look like should suffice.

Reply:
Will add a comment